### ISSUE TITLE
"angle down" icon doesn't work if floated right

### ISSUE THREAD

**Issue by esmondb** [fomantic/Fomantic-UI#739]
# Bug Report

In a menu item an angle down right floated icon comes out as a angle right icon.

## Steps to reproduce
`<i class="angle down right floated icon"></i>`

## Expected result
angle down icon

## Actual result
angle right icon

## Testcase
https://jsfiddle.net/c3xrkfm4/2/

**Comment by hammy2899** [comment]
You don't need `right floated` for the icons to appear on the right of your menu items
https://jsfiddle.net/e0L5rsmh/ so this shouldn't be a problem in this case.

**Comment by esmondb** [comment]
My actual use case is in a dropdown menu. see:
https://jsfiddle.net/c3xrkfm4/3/
Is there a solution for this case?

**Comment by hammy2899** [comment]
@esmondb Sadly there isn't the bug is due to the `right floated` and `right angle` conflicting and the only way to solve it is to make `right angle` have a order specific CSS definition which is something I want to do in v3 but we can't do that right now because that is a breaking change.

I suggest trying to float the icon with some custom CSS.

**Comment by hammy2899** [comment]
@esmondb Sadly there isn't the bug is due to the `right floated` and `right angle` conflicting and the only way to solve it is to make right angle have a order specific CSS definition which is something I want to do in v3 but we can't do that right now because that is a breaking change.

I suggest trying to float the icon with some custom CSS.

**Comment by esmondb** [comment]
Okay, I'll try some custom CSS.
Thanks

**Comment by hammy2899** [comment]
Leave the issue open because it isn't resolved yet 😉

**Comment by lubber-de** [comment]
It is solvable without a breaking change by changing the order in the icon.overrides file from

```less
i.icon.angle.down:before { content: "\f107"; }
i.icon.angle.left:before { content: "\f104"; }
i.icon.angle.right:before { content: "\f105"; }
i.icon.angle.up:before { content: "\f106"; }
```
to
```less
i.icon.angle.left:before { content: "\f104"; }
i.icon.angle.right:before { content: "\f105"; }
i.icon.angle.up:before { content: "\f106"; }
i.icon.angle.down:before { content: "\f107"; }
```
But as the overrides file is actually generated out of the font-awesome lib alphabetically, you might do that on your own and compile FUI again

Possible other workaround by increasing specificity in custom CSS using default FUI package:

```less
i.icon.angle.down.down:before {
    content: "\f107";
}
```
See https://jsfiddle.net/y1jz3uga/

**Comment by esmondb** [comment]
Thanks for the example.

**Comment by exoego** [comment]
Closing since it is released in 2.7.7 https://github.com/fomantic/Fomantic-UI/releases/tag/2.7.7
